lib/internal/child_process/serialization.js

```diff
@@ -51,29 +52,51 @@ class ChildProcessDeserializer extends v8.DefaultDeserializer {
 //   (aka 'advanced')
 const advanced = {
   initMessageChannel(channel) {
-    channel[kMessageBuffer] = Buffer.alloc(0);
+    channel[kMessageBuffer] = [];
+    channel[kMessageBufferSize] = 0;
     channel.buffering = false;
   },
 
   *parseChannelMessages(channel, readData) {
     if (readData.length === 0) return;
 
-    let messageBuffer = Buffer.concat([channel[kMessageBuffer], readData]);
-    while (messageBuffer.length > 4) {
-      const size = messageBuffer.readUInt32BE();
-      if (messageBuffer.length < 4 + size) {
-        break;
-      }
+    channel[kMessageBuffer].push(readData);
+    channel[kMessageBufferSize] += readData.length;
+
+    // Index 0 should always be present because we just pushed data into it.
+    let messageBufferHead = channel[kMessageBuffer][0];
+    while (messageBufferHead.length >= 4) {
+      // We read the uint manually here, because this is faster than first converting
+      // it to a buffer and using `readUInt32BE` on that.
+      const size =
+        messageBufferHead[0] << 24 |
+        messageBufferHead[1] << 16 |
+        messageBufferHead[2] << 8 |
+        messageBufferHead[3];
```

Comment:
```suggestion
      const size = Buffer.prototype.readUInt32BE.call(messageBufferHead, 0) + 4;
```

This might work as well? I added the plus 4 as well. That way there's no need to add it later on. This might require a new variable name as well.

Reply:
I have added these suggestions, since I can't see any noticible performance impact of this after some testing, and it makes the code easier to read.